File: tests/Issue12.hs
#!/usr/bin/env cabal

{- cabal:

  build-depends: base
               , wave

-}

--------------------------------------------------

{-# LANGUAGE PackageImports #-}

--------------------------------------------------

import qualified "wave" Codec.Audio.Wave as W

--------------------------------------------------

main = do

  putStrLn "--------------------------------------------------"
  putStrLn "Reading WAV (Signed PCM)"
  putStrLn "--------------------------------------------------"
  
  W.readWaveFile "./issue-12.wav.pcm"

  putStrLn "--------------------------------------------------"
  putStrLn "Reading WAV (MS ADPCM)"
  putStrLn "--------------------------------------------------"

  W.readWaveFile "./issue-12.wav"

  putStrLn "--------------------------------------------------"
  putStrLn "DONE"
  putStrLn "--------------------------------------------------"

--------------------------------------------------

-- https://github.com/mrkkrp/wave/issues/11
--
-- « readWaveFile » throws « NonPcmFormatButMissingFact » on « ./issue-11.wav ».
--

--------------------------------------------------

File: tests/issue_12.py
#!/usr/bin/env python
##################################################

import wave

##################################################

with wave.open("./issue-12.wav", "rb") as audio:

    print audio.getnframes()

##################################################

# <<< wave.Error: unknown format: 2 >>>

# Python's wave module only works with one ype of WAV: PCM (WAVE_FORMAT_PCM: 0x0001).
# "./issue-12.wav" is a WAV of type Adaptive Differential PCM (WAVE_FORMAT_ADPCM: 0x0002)

##################################################